easy-ui-react/src/Select/SelectField.tsx

```diff
@@ -0,0 +1,133 @@
+import { SelectTrigger } from "./SelectTrigger";
+
+export type SelectFieldSize = "sm" | "md" | "lg";
+export type ValidationState = "valid" | "invalid";
+
+export type SelectFieldProps = {
+  /**
+   * Visually hides the label, but keeps it accessible.
+   * @default false
+   */
+  isLabelVisuallyHidden?: boolean;
+  /**
+   * Whether the select field is disabled.
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether user input is required on the select field before form submission.
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * Whether the input should display its "valid" or "invalid" visual styling.
+   * @default valid
+   */
+  validationState?: ValidationState;
+  /**
+   * Label text displays with emphasis.
+   * @default false
+   */
+  isLabelEmphasized?: boolean;
+  /**
+   * Size affects the overall size of the select field, but it also influences
+   * the size of iconAtStart.
+   * @default md
+   */
+  size?: SelectFieldSize;
+  /** The content to display as the label. */
+  label: ReactNode;
+  /** Error text that appears below select field. */
+  errorText?: ReactNode;
+  /** Helper text that appears below select field. */
+  helperText?: ReactNode;
+  /** Temporary text that occupies select field when it is empty. */
+  placeholder?: string;
+  /** Left aligned icon on input. */
+  iconAtStart?: IconSymbol;
+  /** Label props associated with field. */
+  labelProps?: DOMAttributes<FocusableElement>;
+  /** Helper text props associated with field. */
+  helperTextProps?: DOMAttributes<FocusableElement>;
+  /** Error text props associated with field. */
+  errorTextProps?: DOMAttributes<FocusableElement>;
+  /** Field value props. */
+  valueProps?: DOMAttributes<FocusableElement>;
+};
+
+export function SelectField(props: SelectFieldProps) {
+  const {
+    isLabelVisuallyHidden = false,
+    isDisabled = false,
+    validationState = "valid",
+    isLabelEmphasized = false,
+    size = "md",
+    label,
+    errorText,
+    helperText,
+    placeholder,
+    iconAtStart,
+    labelProps,
+    valueProps,
+    errorTextProps,
+    helperTextProps,
+  } = props;
+  const { triggerRef, selectState } = useInternalSelectContext();
+
+  const hasError = validationState === "invalid";
+  const showErrorText = hasError && errorText;
+  const showHelperText = !showErrorText && helperText;
+  const captionProps = showHelperText ? helperTextProps : errorTextProps;
+  const captionText = showHelperText ? helperText : errorText;
+
+  return (
+    <div className={classNames(styles.fieldRoot)}>
+      <Label
+        isLabelVisuallyHidden={isLabelVisuallyHidden}
+        fieldSize={size}
+        hasError={hasError}
+        isLabelEmphasized={isLabelEmphasized}
+        {...labelProps}
+      >
+        {label}
+      </Label>
+      <HiddenSelect
+        isDisabled={isDisabled}
+        state={selectState}
+        triggerRef={triggerRef}
+        label={label}
+      />
```

Comment:
react-aria's `HiddenSelect` component is causing an accessibility violation according to storybook:

![sb_violation](https://github.com/EasyPost/easy-ui/assets/44451175/90be1aa2-109b-42a9-a709-fde7d56080b6)

according to react-aria, changing the `visibility` to `hidden` would break some functionality on safari: https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/select/src/HiddenSelect.tsx#L53